prog/download_boot_image.rb

```diff
@@ -10,13 +10,30 @@ def image_name
   end
 
   def version
-    @version ||= frame.fetch("version")
+    @version ||= frame.fetch("version") { default_boot_image_version(image_name) }
```

Comment:
Ruby calculates the default value even if the fetch returns some value. When the block is passed it only calculates the value if the fetch doesn't return. I wanted to keep the tests as it is, so block is used instead of default value directly.